### PR TITLE
Integrate functions

### PR DESCRIPTION
**Pull request by amalexandrescu**
Starting to integrate the functionalities on frontend side

### CODE REVIEW

File: src/components/ConnectWallet.tsx [hunk 1]
```diff
@@ -15,19 +16,33 @@ interface ConnectWalletProps {
...
     if (location.pathname !== '/') {
-      if (location.pathname.substring(1).toLowerCase() !== currentRole.toLowerCase()) {
+      if (location.pathname.substring(1)?.toLowerCase() !== currentRole.toLowerCase()) {
         console.log('Seems like you got lost, click here to go back to the main page');
       }
     }
   };
+  useEffect(() => {
+    const fetchStatus = async () => {
+      const args = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/ConnectWallet.tsx:35]: this should take the address based on the network from const file.

mainnet => .mainnet
testnet/devnet => .testnet

File: src/components/ConnectWallet.tsx [hunk 2]
```diff
@@ -43,7 +58,8 @@ const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
 
   if (userSession.isUserSignedIn()) {
     if (currentRole === 'Viewer') {
-      dispatch(updateUserRoleAction());
+      dispatch(updateUserRoleAction(finalStatus));
+      // dispatch(updateUserRoleAction());
```
Comment (BowTiedDeployer) [inline src/components/ConnectWallet.tsx:62]: remove this

File: src/components/appMenuSections/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -37,6 +47,37 @@ const Dashboard = () => {
...
+      const notifier = await readOnlyGetNotifier();
+      setCurrentNotifier(notifier);
+    };
+
+    getCurrentNotifier();
+  }, [currentNotifier]);
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      const args = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/dashboard/Dashboard.tsx:61]: declare at beginning networkUsed = mainnet if config-network is mainnet, else testnet ( for testnet and devnet ).

don't know if this is not fixed in a future pr. @BowTiedJohn is it?

**Review by BowTiedDeployer — APPROVED** [review]

File: src/components/appMenuSections/dashboard/Dashboard.tsx (hunk 3 above)
Comment (BowTiedJohn) [reply src/components/appMenuSections/dashboard/Dashboard.tsx:61]: I don't think it is, but shouldn't we put it on Header Bar instead of Dashboard?

Comment (BowTiedDeployer) [reply src/components/appMenuSections/dashboard/Dashboard.tsx:61]: it is not changed in a future opened pr. can be done in a future commit.

File: src/components/ConnectWallet.tsx (hunk 1 above)
Comment (BowTiedDeployer) [reply src/components/ConnectWallet.tsx:35]: it is not changed in a future opened pr. can be done in a future commit.

**Comment by BowTiedDeployer** [comment]
This should be done as the last PR, after everything is merged together, then in `Master`.

File: src/components/ConnectWallet.tsx [hunk 4]
```diff
@@ -15,19 +16,33 @@ interface ConnectWalletProps {
...
+      if (location.pathname.substring(1)?.toLowerCase() !== currentRole.toLowerCase()) {
         console.log('Seems like you got lost, click here to go back to the main page');
       }
     }
   };
+  useEffect(() => {
+    const fetchStatus = async () => {
+      const args = userSession.loadUserData().profile.stxAddress.testnet;
+      const status = await readOnlyAddressStatus(args);
+      console.log('args', args);
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx:37]: Cleanup

File: src/components/ConnectWallet.tsx [hunk 5]
```diff
@@ -15,19 +16,33 @@ interface ConnectWalletProps {
...
+      setFinalStatus(status);
+
+      updateUserRoleAction(finalStatus);
+      console.log('status', finalStatus);
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx:41]: Cleanup

File: src/components/appMenuSections/dashboard/Dashboard.tsx [hunk 6]
```diff
@@ -49,42 +90,61 @@ const Dashboard = () => {
...
-          Join pool
-        </Button>
-      )}
-      {clickedJoinPoolButtonByViewer && authenticatedSuccessfully === false && (
-        <LoadingButton loading loadingPosition="start" startIcon={<SaveIcon />} variant="outlined">
-          JOIN POOL
-        </LoadingButton>
-      )}
-    </div>
+    <Box
```
Comment (Alexandrescu) [inline src/components/appMenuSections/dashboard/Dashboard.tsx:93]: Created a ticket for this https://github.com/stacks-degens/starters-front-end/issues/44

File: src/components/appMenuSections/miningPool/MiningPool.tsx [hunk 7]
```diff
@@ -3,119 +3,56 @@ import TableCell from '@mui/material/TableCell';
...
-      ContractVoteNegativeJoin(address);
-    }
-  };
+  }, [finalStatus]);
 
   const handleMinerRemoveButtonClick = (address: string) => {
     ContractProposeRemoval(address);
   };
 
   const handleMinerInfoButtonClick = (address: string) => {
```
Comment (Alexandrescu) [inline src/components/appMenuSections/miningPool/MiningPool.tsx:52]: Not used

**Review by Alexandrescu — APPROVED** [review]
I created tickets for what needs fixed in this PR. We also need to do a final pass and get all of the components cleaned up.

**Pull request merged**